src/Math/Geometry/Rectangle.cs

```diff
@@ -0,0 +1,162 @@
+
+        /// <summary>
+        /// The y-coordinate of the top boundary of the rectangle.
+        /// </summary>
+        public float Top
+        {
+            get { return this.y; }
+        }
+
+        /// <summary>
+        /// The y-coordinate of the bottom boundary of the rectangle.
+        /// </summary>
+        public float Bottom
+        {
+            get { return this.y + this.height; }
+        }
+
+        /// <summary>
+        /// The coordinates of the top left corner of the rectangle.
+        /// </summary>
+        public Vector2 TopLeft
+        {
+            get { return new Vector2(this.Left, this.Top); }
+        }
+
+        /// <summary>
+        /// The coordinates of the top right corner of the rectangle.
+        /// </summary>
+        public Vector2 TopRight
+        {
+            get { return new Vector2(this.Right, this.Top); }
+        }
+
+        /// <summary>
+        /// The coordinates of the bottom left corner of the rectangle.
+        /// </summary>
+        public Vector2 BottomLeft
+        {
+            get { return new Vector2(this.Left, this.Bottom); }
+        }
+
+        /// <summary>
+        /// The coordinates of the bottom right corner of the rectangle.
+        /// </summary>
+        public Vector2 BottomRight
+        {
+            get { return new Vector2(this.Right, this.Bottom); }
+        }
+        
+        /// <summary>
+        /// The coordinates of the center of mass of the rectangle.
+        /// </summary>
+        public Vector2 Center
+        {
+            get { return new Vector2(this.x + this.width * 0.5f, this.y + this.height * 0.5f); }
+        }
+        #endregion
+
+        /// <summary>
+        /// Creates a new rectangle.
+        /// </summary>
+        /// <param name="x">X-coordinate of the upper left corner.</param>
+        /// <param name="y">Y-coordinate of the upper left corner.</param>
+        /// <param name="width">Width of the rectangle.</param>
+        /// <param name="height">Height of the rectangle.</param>
+        public Rectangle(float x, float y, float width, float height)
+        {
+            this.x = x;
+            this.y = y;
+            this.width = width;
+            this.height = height;
+        }
+
+        /// <summary>
+        /// Creates a new rectangle.
+        /// </summary>
+        /// <param name="xy">Coordinates of the upper left corner.</param>
+        /// <param name="width">Width of the rectangle.</param>
+        /// <param name="height">Height of the rectangle.</param>
+        public Rectangle(Vector2 xy, float width, float height)
+            : this(xy.X, xy.Y, width, height) { }
+
+        /// <summary>
+        /// Creates a new rectangle.
+        /// </summary>
+        /// <param name="upperLeft">Coordinates of the upper left corner.</param>
+        /// <param name="bottomRight">Coordinates of the bottom right corner.</param>
+        public Rectangle(Vector2 upperLeft, Vector2 bottomRight)
```

Comment:
This may be confused with a possible Rectangle(Vector2 upperLeft, Vector2 size).

Consider using static methods for these constructurs instead?